identity/app/controllers/EditProfileController.scala

```diff
@@ -49,6 +50,24 @@ class EditProfileController(
   def displayDigitalPackForm: Action[AnyContent] = displayForm(DigiPackEditProfilePage)
   def displayEmailPrefsForm: Action[AnyContent] = displayForm(EmailPrefsProfilePage)
 
+  def displayRepermissioningJourneyForm: Action[AnyContent] = {
+    if (IdentityAllowAccessToGdprJourneyPageSwitch.isSwitchedOff) {
+      recentlyAuthenticated { implicit request =>
+        NotFound(views.html.errors._404())
+      }
+    }
+    else {
+      csrfAddToken {
+        recentlyAuthenticated.async { implicit request =>
+          repermissionJourneyView(
+            page = EmailPrefsProfilePage,
```

Comment:
i've added a bunch of gibberish to have all possible ground covered but this could be trimmed down later